kura/org.eclipse.kura.core.configuration/src/main/java/org/eclipse/kura/core/configuration/ConfigurationServiceImpl.java

```diff
@@ -1312,7 +1333,11 @@ XmlComponentConfigurations loadEncryptedSnapshotFileContent(long snapshotID) thr
         }
 
         // File loaded, try to decrypt and unmarshall
-        String decryptedContent = new String(this.m_cryptoService.decryptAes(entireFile.toString().toCharArray()));
+        char[] decryptAes = this.m_cryptoService.decryptAes(entireFile.toString().toCharArray());
+        if (decryptAes == null) {
```

Comment:
I agree with @markoer Unless Kura is full of assumptions which turn into NPEs every now and then. So checking for NULL doesn't hurt, unless Kura would adopt proper null pointer analysis.